### PR TITLE
profile implemented

### PR DESCRIPTION
**Pull request by batchatT**

### CODE REVIEW

File: app/src/main/java/com/example/trempelapp/presentation_layer/main_screen/main_fragments/profile/ProfilePageViewModel.kt [hunk 1]
```diff
@@ -22,4 +37,45 @@ class ProfilePageViewModel @Inject constructor(
...
+            })
+            .run(compositeDisposable::add)
+    }
+
+    private fun writeUserImageUri(uri: Uri) {
+        writeUserImageUri
+            .execute(uri)
+            .subscribeOn(Schedulers.io())
+            .observeOn(AndroidSchedulers.mainThread())
+            .subscribe({}, {})
```
Comment (denisque75) [inline app/src/main/java/com/example/trempelapp/presentation_layer/main_screen/main_fragments/profile/ProfilePageViewModel.kt:71]: why success and error methods are empty?

File: data/src/main/java/com/example/data/shared_preferences/SharedPreferencesManagerImpl.kt [hunk 2]
```diff
@@ -2,12 +2,16 @@ package com.example.data.shared_preferences
...
 import android.content.Context
 import android.content.SharedPreferences
+import androidx.core.net.toUri
+import com.example.domain_layer.models.AdditionalUserInfo
 import com.example.trempelapp.utils.EMPTY_STRING
 import com.example.trempelapp.utils.TOKEN_PREFERENCES_FILE
 import io.reactivex.Single
 import javax.inject.Inject
 
 private const val TOKEN = "token"
```
Comment (denisque75) [inline data/src/main/java/com/example/data/shared_preferences/SharedPreferencesManagerImpl.kt:12]: make those constants as companion object

File: data/src/main/java/com/example/data/shared_preferences/SharedPreferencesManagerImpl.kt [hunk 3]
```diff
@@ -21,13 +25,35 @@ class SharedPreferencesManagerImpl @Inject constructor(
...
+    override fun getAdditionalUserInfo(): Single<AdditionalUserInfo> {
+        val data = AdditionalUserInfo(
+            login = preferences.getString(LOGIN, EMPTY_STRING) ?: EMPTY_STRING,
+            imageUri = preferences.getString(IMAGE, EMPTY_STRING)?.toUri() ?: EMPTY_STRING.toUri()
+        )
+        return Single.just(data)
+    }
+
     override fun writeToken(token: String) {
         val editor = preferences.edit()
```
Comment (denisque75) [inline data/src/main/java/com/example/data/shared_preferences/SharedPreferencesManagerImpl.kt:37]: let's create private methods here something like `saveValue(key, value)`
In those methods, you will do the magic like 
`editor
            .putString(TOKEN, token)
            .apply()` and save other data
F.E. your `writeUserLogin(...)` will looks like
`override fun writeUserLogin(login: String) {
        saveString(LOGIN, login)
    }`

File: data/src/main/java/com/example/data/shared_preferences/SharedPreferencesManagerImpl.kt (hunk 2 above)
Comment (batchatT) [reply data/src/main/java/com/example/data/shared_preferences/SharedPreferencesManagerImpl.kt:12]: done

File: data/src/main/java/com/example/data/shared_preferences/SharedPreferencesManagerImpl.kt (hunk 3 above)
Comment (batchatT) [reply data/src/main/java/com/example/data/shared_preferences/SharedPreferencesManagerImpl.kt:37]: done

File: app/src/main/java/com/example/trempelapp/presentation_layer/main_screen/main_fragments/profile/ProfilePageViewModel.kt (hunk 1 above)
Comment (batchatT) [reply app/src/main/java/com/example/trempelapp/presentation_layer/main_screen/main_fragments/profile/ProfilePageViewModel.kt]: done

**Review by denisque75 — APPROVED** [review]

**Pull request merged**